src/dfcx_scrapi/tools/analysis_util.py

```diff
@@ -0,0 +1,162 @@
+"""Util class for performing analysis on DFCX objects and data."""
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import pandas
+import numpy as np
+from dfcx_scrapi.core.intents import Intents
+import google.cloud.dialogflowcx_v3beta1.types as types
+
+# logging config
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s %(levelname)-8s %(message)s',
+    datefmt='%Y-%m-%d %H:%M:%S')
+
+class AnalysisUtils():
+    """Utils class for quickly building DFCX webhook payloads."""
+
+    @staticmethod
+    def levenshtein_ratio(s, t):
+        """ levenshtein_ratio_and_distance:
+            Calculates levenshtein distance between two strings.
+        """
+        rows = len(s)+1
+        cols = len(t)+1
+        distance = np.zeros((rows,cols),dtype = int)
+
+        for i in range(1, rows):
+            for k in range(1,cols):
+                distance[i][0] = i
+                distance[0][k] = k
+
+        for col in range(1, cols):
+            for row in range(1, rows):
+                if s[row-1] == t[col-1]:
+                    cost = 0
+                else:
+                    cost = 2
+
+                distance[row][col] =  min(
+                    distance[row-1][col] + 1,
+                    distance[row][col-1] + 1,
+                    distance[row-1][col-1] + cost
+                )
+
+        # Computation of the Levenshtein Distance Ratio
+        ratio = ((len(s)+len(t)) - distance[row][col]) / (len(s)+len(t))
+        return ratio
+
+
+    @staticmethod
```

Comment:
@Omerside since this is calling `levenshtein_ratio` consider changing this to an instance method instead of a static method. 

From looking at this Class, I think the assumption here is that the user would directly call `calc_tp_distances()` but never really call `levenshtein_ratio()` directly. Is that correct?

Reply:
Correct. I'll change that.

Reply:
Changing it to an instance method creates a weird situation when trying to invoke it from a static method, since there's no "self" defined. I've designated it as private by means of a double-underscore prefix. Will this suffice?

Reply:
@Omerside any issue with including `self` ?
So that you can also change the line [here](https://github.com/GoogleCloudPlatform/dfcx-scrapi/pull/5/files#diff-a21070c47535c8b96b766ff6fcc27fa6b0a09db27dc11589e391cdeb726f6bc8R142) to:

```
distance = self.levenshtein_ratio(....)
```

Reply:
Thanks for adding the sample code. This structure makes sense now.